src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -208,11 +283,93 @@ public static class AccessTokenInfo {
+                    if(Arrays.stream(protectedPropNames).noneMatch(element.getKey()::equals)){
+                        userData.add(element.getKey(), element.getValue());
+                    }
+                }
+
+                return new AccessTokenInfo(
+                        payload.get("sessionHandle").getAsString(),
+                        payload.get("sub").getAsString(),
+                        payload.get("refreshTokenHash1").getAsString(),
+                        payload.get("exp").getAsLong() * 1000,
+                        parentRefreshTokenHash.isJsonNull() ? null : parentRefreshTokenHash.getAsString(),
+                        userData,
+                        antiCsrfToken.isJsonNull() ? null : antiCsrfToken.getAsString(),
+                        payload.get("iat").getAsLong() * 1000,
+                        version
+                );
+            } else {
+                checkRequiredPropsExist(payload, requiredPropsV2);
+                return new AccessTokenInfo(
+                        payload.get("sessionHandle").getAsString(),
+                        payload.get("userId").getAsString(),
+                        payload.get("refreshTokenHash1").getAsString(),
+                        payload.get("expiryTime").getAsLong(),
+                        parentRefreshTokenHash.isJsonNull() ? null : parentRefreshTokenHash.getAsString(),
+                        payload.get("userData").getAsJsonObject(),
+                        antiCsrfToken.isJsonNull() ? null : antiCsrfToken.getAsString(),
+                        payload.get("timeCreated").getAsLong(),
+                        version
+                );
+            }
+
+        }
+
+        JsonObject toJSON() throws UnauthorisedException {
+            JsonObject res = new JsonObject();
+            if (this.version == VERSION.V3) {
+                res.addProperty("sub", this.userId);
+                res.addProperty("exp", this.expiryTime / 1000);
+                res.addProperty("iat", this.timeCreated / 1000);
+            } else {
+                res.addProperty("userId", this.userId);
+                res.addProperty("expiryTime", this.expiryTime);
+                res.addProperty("timeCreated", this.timeCreated);
+            }
+            res.addProperty("sessionHandle", this.sessionHandle);
+            res.addProperty("refreshTokenHash1", this.refreshTokenHash1);
+            res.addProperty("parentRefreshTokenHash1", this.parentRefreshTokenHash1);
+            res.addProperty("antiCsrfToken", this.antiCsrfToken);
+
+            if (this.version == VERSION.V3) {
+                for (Map.Entry<String, JsonElement> element : this.userData.entrySet()) {
+                    if (res.has(element.getKey())) {
+                        // The use is trying to add a protected prop into the payload (userId, etc) in V3 (this should be blocked by validation)
+                        throw new UnauthorisedException("The user payload contains protected field");
```

Comment:
Create a new type of exception for this cause UnauthorisedException is meant for when the session is used, but it deosn't exist, or if the refresh token is invalid etc.